### DISCUSSION TITLE
Integration of an option aimed at adjusting the screen resolution

### DISCUSSION THREAD

**Discussion by Ricky-Tigg** [rhinstaller/anaconda#4746] [General]
Hello. The adequate **font size**, and for what is relevant here and related, subsequently the **displayed screen resolution**, is if any, what makes a very reading content to come to existence, by making it visible, thus readable, for a given eye. Since the beginning of an OS manual installation process invariably starts from the boot of the Anaconda installation media, that is from that very moment a content exhibited should be readable.

As it appears to be at Anaconda's boot, a **default display resolution is applied to content exhibited**, which is within a graphical environment. Since that resolution matches the highest one a screen is capable of, as a direct consequence, the highest that resolution is, the smallest the font sizes of exhibited content are.

_As context_ | Unlike other Fedora releases, the _Desktop_ edition activates Gnome desktop environment in Anaconda, which is then reachable by the person performing the manual installation. **As a consequence of that convenience, the resolution can be set during the installation session and takes effect on-the-fly.**

For a given eye, assumed mine, how visible would be some of the texts with smaller font sizes on a **15.6 inches** laptop whose the highest resolution its screen is capable of is **1920 x 1080 (16:9)**? The current content would not be visible. Barely visible with 1600 x 900 (16:9) set. At last, it would be with 1368 x 768 (16:9) set.

A visit at the documentation [page](https://anaconda-installer.readthedocs.io/en/latest/index.html) along with a search against "_resolution_" shall
reveal the obvious; nothing relevant in this regard. Then we know how it is going to end, as we can easily guess what could reasonably be expected to happen in the same condition when the OS to install lacks that convenience. For instance a such OS can be one of those other editions that were previously referred to. **That would most likely lead to make an installation improbable.** When thinking of the upstream effort. All the lost effort, due to the incapacity to adjust internally by any mean a non-adequate resolution. Isn't it a pity?

**Comment by VladimirSlavik** [comment]
Would this help? https://anaconda-installer.readthedocs.io/en/latest/boot-options.html#inst-resolution

(As we're moving towards WebUI, eventually you might be able to just zoom the view...)

**Comment by Ricky-Tigg** [comment]
It does exist indeed. I made a weak investigation. I conducted a test in a virtual machine from an ISO source file. Here are the relevant information:

- Physical machine
  - A Linux OS installed
  - Current resolution: **1368 x 768 (16:9)**
  - Screen physical size:
```
$ monitor-get-edid | monitor-parse-edid | grep -i '^screen'
Screen size: 34.4 cm x 19.3 cm (15.53 inches, aspect ratio 16/9 = 1.78)
```
_Note:_ size values verified with a physical ruler.

- Virtual machine
  - As prerequisite | Investigation of  resolutions supported by the enabled virtual video driver (tested _Virtio_ and _VGA_); _1360 x 768 (16:9)_ is.
  - Runs as qcow2 disk image in QEMU-KVM on `virt-manager` platform.
  - OS: Fedora Desktop edition
  - Machine displayed in full screen and scaled

Added parameter among others in GRUB v.2.0.6 session: `inst.resolution=1360x768`

Once started, the current resolution reveals to be **1280 x 800 (16:10)**,  thus the query related to the resolution is not honored.
